hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -252,66 +256,76 @@ func (a OpenStack) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, _
 							},
 						},
 					},
-				}, {
-					Name:            "orc-manager",
-					Image:           orcImage,
-					ImagePullPolicy: corev1.PullIfNotPresent,
-					Command:         []string{"/manager"},
-					Args: []string{
-						"--namespace=$(MY_NAMESPACE)",
-						"--leader-elect",
-						"--health-probe-bind-address=:8081",
-					},
-					SecurityContext: &corev1.SecurityContext{
-						AllowPrivilegeEscalation: &allowPrivilegeEscalation,
-						Capabilities: &corev1.Capabilities{
-							Drop: []corev1.Capability{
-								"ALL",
-							},
-						},
-					},
-					LivenessProbe: &corev1.Probe{
-						ProbeHandler: corev1.ProbeHandler{
-							HTTPGet: &corev1.HTTPGetAction{
-								Path: "/healthz",
-								Port: intstr.FromInt(8081),
-							},
-						},
-						InitialDelaySeconds: 15,
-						PeriodSeconds:       20,
+				}},
+			}},
+	}
+
+	// Add the ORC manager container if the payload version is 4.19 or later
+	// ORC was decoupled from CAPO in 4.19 but was part of CAPO in 4.18.
+	if a.payloadVersion != nil && a.payloadVersion.Major == 4 && a.payloadVersion.Minor >= 19 {
```

Comment:
This would benefit from unit tests that validate the spec based on the passed payloadVersion